cpp-ch/local-engine/Functions/SparkFunctionCheckDecimalOverflow.cpp

```diff
@@ -121,7 +121,17 @@ class FunctionCheckDecimalOverflow : public IFunction
 
                 if (const ColVecType * col_vec = checkAndGetColumn<ColVecType>(src_column.column.get()))
                 {
-                    executeInternal<FromFieldType, ToDataType>(*col_vec, result_column, input_rows_count, precision, scale);
+                    executeInternal<FromDataType, ToDataType, ColVecType>(*col_vec, result_column, input_rows_count, precision, scale);
+                    return true;
+                }
+            }
+            else if constexpr (IsDataTypeNumber<FromDataType>)
```

Comment:
above if and else if should be merged. Remind: use `using ColVecType = ColumnVectorOrDecimal<T>;
`